test/e2e/atlas/setup_failure_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e || (atlas && interactive)
+
+package atlas_test
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongodb-atlas-cli/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetupFailureFlow(t *testing.T) {
+	g := newAtlasE2ETestGenerator(t)
+	g.generateProject("setup")
+	cliPath, err := e2e.AtlasCLIBin()
+	req := require.New(t)
+	req.NoError(err)
+
+	t.Run("Invalid Public Key", func(t *testing.T) {
+		t.Setenv("MCLI_PUBLIC_API_KEY", "invalid_public_key")
```

Comment:
I am wondering if changing this env would have an effect also on other tests 🤔 Could we store the original value with `GetEnv` and then use again `SetEnv` at the end of the test to set the original value?

Reply:
Here is your answer:

```go
func (*testing.T).Setenv(key string, value string)
Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.
```

Reply:
https://pkg.go.dev/testing#T.Setenv

>Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.

Reply:
Thanks!